File: VB/WcfService1/Service1.svc.vb
Imports DevExpress.Xpo

Imports DevExpress.Xpo.DB
Imports System.ServiceModel

Namespace WcfService1

    Public Class Service1
        Inherits DataStoreService

        Public Shared DataStore As IDataStore

        Shared Sub New()

            Dim connectionString As String = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB")

            DataStore = XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema)

        End Sub
        Public Sub New()
            MyBase.New(DataStore)
        End Sub

    End Class

End Namespace

File: CS/WcfService1/Service1.svc.cs
using DevExpress.Xpo;

using DevExpress.Xpo.DB;
using System.ServiceModel;

namespace WcfService1 {

    public class Service1 : DataStoreService {

        public static IDataStore DataStore;

        static Service1() {

            string connectionString = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB");

            DataStore = XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema);
            
        }
        public Service1()

            : base(DataStore) {
        }

    }

}
